python/torch_mlir/eager_mode/torch_mlir_dispatch.py

```diff
@@ -125,7 +126,11 @@ def build_script_function(
     else:
         graph.registerOutput(node.output())
 
-    fn_name = str(node).strip()
+    # The first alternation handles non-alpha characters at the beginning.
+    # MLIR doesn't support such symbol names unless quoted, hence safer to remove.
+    # The second alternation handles non-word characters in the rest of the name (eg =, :, etc).
+    # LLVM doesn't support such symbol names.
+    fn_name = re.sub(r'^[^a-zA-Z]+|[\W_]+', '', str(node))
```

Comment:
I'm not sure I understand the requirements here. In what way is this name load-bearing? Can we just call it "f"? If not, then I think we also need to mangle the tensor shapes we specialized for as well.

Reply:
it's load-bearing because we use this name to pull the exported function out of the module here

https://github.com/llvm/torch-mlir/blob/9ec0683e928bdd1acea9aeb025ed542332fab902/python/torch_mlir/eager_mode/torch_mlir_dispatch.py#L260

>op_mlir_backend_callable = getattr(loaded_module, script_fun.name)

right now there's only ever one exported function in that module but in the future there could be others (in the case that we decide to incrementally compile and inline or not).

>If not, then I think we also need to mangle the tensor shapes we specialized for as well.

good idea!

Reply:
Looking at the code, it seems like it would still continue to work if we just called the function "f" every time. Right? That seems simplest to do so I think we should do it.